gitlab/_backends/protocol.py

```diff
@@ -0,0 +1,34 @@
+import abc
+import sys
+from typing import Any, Dict, Optional, Union
+
+import requests
+from requests_toolbelt.multipart.encoder import MultipartEncoder  # type: ignore
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
```

Comment:
Actually, just one more comment. This won't work on 3.7 without a change in `setup.py`. But I'll open a follow-up for that.